arm-appinsights/2015-05-01/swagger/component_API.json

```diff
@@ -0,0 +1,376 @@
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponent"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request when creating or updating an Application Insights component. The updated component is returned.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponent"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ComponentCreate": { "$ref": "../examples/ComponentCreate.json" },
+          "ComponentUpdate": { "$ref": "../examples/ComponentUpdate.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "ApplicationInsightsComponent": {
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "The kind of application that this component refers to, used to customize UI. This value is a freeform string, values should typically be one of the following: web, ios, other, store, java, phone."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationInsightsComponentProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "required": [
+        "kind"
+      ],
+      "description": "An Application Insights component definition."
+    },
+    "ApplicationInsightsComponentProperties": {
+      "description": "Properties that define an Application Insights component resource.",
+      "properties": {
+        "ApplicationId": {
+          "type": "string",
+          "description": "The unique ID of your application."
+        },
+        "AppId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Application Insights Unique ID for your Application."
+        },
+        "Application_Type": {
+          "type": "string",
+          "description": "Type of application being monitored. Example values: Web, other. Cannot be changed after the component has been created."
+        },
+        "Flow_Type": {
```

Comment:
There is a way to make a model property or a parameter a constant. You should mark it as "required" and add an enum constraint `"enum": ["Bluefield"]` with one value. Autorest will treat this as a constant and will not expose it. It will always serialize that property/parameter with the specified value.

One thing that I am confused about, from your comment
> and leave it that way **no matter how the user sets it?**

If you make something a constant the way I specified above, then the user will never be able to set it. It will always be "Bluefield" on the wire.